scalafmt-tests/src/test/resources/scala3/OptionalBraces.stat

```diff
@@ -275,3 +275,57 @@ object a {
      end cond
   end A
 }
+<<< match
+enum IndentWidth:
+  def foo =
+    this match
+      case A =>
+         that match
+            case b => bb
+            case c => cc
+         end match
+      case B => // should be at same level as case A
+         that match
+            case c => cc
+            case _ => dd
+>>>
+enum IndentWidth:
+   def foo =
+     this match
+        case A =>
+          that match
+             case b => bb
+             case c => cc
+          end match
+          case B => // should be at same level as case A
```

Comment:
Looks like an issue with the parser, taking a look into it!

Reply:
https://github.com/scalameta/scalameta/pull/2276

Reply:
That's merged, should we release a new version, bump it here with the fix and tag the release candidate? Or maybe did you find some more issues to fix?

Reply:
i am adding as many examples i can find online, to see how this works.

i also tried to implement the rewrite dropping braces, and that's also where some of the issues were detected.

Reply:
ok cool, I will hold off on releasing then

Reply:
i think we could release the parser, hopefully there are no more issues with it. that way, we can test the formatter with the updated version.